src/SourceCode.Clay/StringExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace SourceCode.Clay
+{
+    /// <summary>
+    /// Utility functions for <see cref="System.String"/>.
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Returns the specified number of characters from the left of a string.
+        /// Tolerates <paramref name="length"/> values that are too large or too small (or negative).
+        /// </summary>
+        /// <param name="str">The string.</param>
+        /// <param name="length">The length.</param>
+        /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static string Left(this string str, int length)
+        {
+            var len = length <= 0 ? 0 : length;
+
+            if (str == null || str.Length <= len) return str;
+            return str.Substring(0, len);
+        }
+
+        /// <summary>
+        /// Returns the specified number of characters from the right of a string.
+        /// Tolerates <paramref name="length"/> values that are too large or too small (or negative).
+        /// </summary>
+        /// <param name="str">The string.</param>
+        /// <param name="length">The length.</param>
+        /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static string Right(this string str, int length)
```

Comment:
If you want to mimic SQL Server's RIGHT function, you should return the whole string when its length is less that the supplied length parameter value.

https://docs.microsoft.com/en-us/sql/t-sql/functions/right-transact-sql

Pseudo code:

if(string.IsNullOrEmpty(str)) return str;

var len = Math.Min(length, str.Length);
return str.Substring(str.Length - len, len);